fly/commands/rename_pipeline.go

```diff
@@ -6,23 +6,20 @@ import (
 	"strings"
 
 	"github.com/concourse/concourse/fly/commands/internal/displayhelpers"
-	"github.com/concourse/concourse/fly/commands/internal/flaghelpers"
 	"github.com/concourse/concourse/fly/rc"
 )
 
 type RenamePipelineCommand struct {
-	Pipeline flaghelpers.PipelineFlag `short:"o"  long:"old-name" required:"true"  description:"Pipeline to rename"`
-	NewName  string                   `short:"n"  long:"new-name" required:"true"  description:"Name to set as pipeline name"`
+	OldName string `short:"o"  long:"old-name" required:"true"  description:"Pipeline to rename"`
+	NewName string `short:"n"  long:"new-name" required:"true"  description:"Name to set as pipeline name"`
 }
 
 func (command *RenamePipelineCommand) Validate() error {
-	_, err := command.Pipeline.Validate()
-	if err != nil {
-		return err
+	if strings.Contains(command.OldName, "/") {
+		return errors.New("old pipeline name cannot contain '/'")
 	}
-
 	if strings.Contains(command.NewName, "/") {
-		return errors.New("pipeline name cannot contain '/'")
+		return errors.New("new pipeline name cannot contain '/'")
 	}
 	return nil
 }
```

Comment:
What is the point of checking for just `/` instead of using `atc.ValidateIdentifier`? The atc side does verify that the new name is a valid identifier, if we want to be extra safe we can check the old name here as well.

Reply:
The point of checking for `/` is to make it to make it clear to the user that they can't rename an individual instance. e.g. if they tried to do

```
$ fly rename-pipeline -o old/branch:master -n new/branch:master
```

we'd catch them and indicate that it expects a pipeline name.

Your idea of using `atc.ValidateIdentifier` is interesting, but I'm not sure we want to do that yet. Right now, you're allowed to rename a pipeline to an invalid identifier - you get a warning in the API response, but it doesn't stop you.

```
$ fly -t dev rp -o pipeline -n _illegal-name

DEPRECATION WARNING:
  - pipeline: '_illegal-name' is not a valid identifier: must start with a lowercase letter

identifier schema documentation: https://concourse-ci.org/config-basics.html#schema.identifier

pipeline successfully renamed to '_illegal-name'
```

Eventually, when we impose full restrictions, it could be better UX to just reject the identifier client-side, but for now, `fly` probably shouldn't be more restrictive than the API

That previous statement is a bit contradictory to this `/` validation, but at least we've had that validation for a while in `fly`

Reply:
In that case we should update the help text to be more informative
```
      -o, --old-name=        Pipeline to rename. For instance pipelines, this will be the name of the entire group
      -n, --new-name=        New name for the pipeline or instance group
```

Reply:
Good call, changed it to:

```
      -o, --old-name=        Existing pipeline or instance group to rename
      -n, --new-name=        New name for the pipeline or instance group
```

Worth noting that instanced pipelines are hidden away in `fly` currently since the feature is still experimental - we don't even show the `--instance-var` flag in the help text. But it's probably still a good idea to document this here since the behaviour isn't obvious